tests/model/test_model.py

```diff
@@ -337,6 +337,14 @@ def test_statements():
     assign = Assignment.create(Expr.function('A', Expr.symbol('t')), Expr.symbol('t'))
     Model.create('model', statements=Statements() + assign)
 
+    assign = Assignment.create(Expr.function('A', 0), Expr.symbol('t'))
+    with pytest.raises(ValueError, match='Symbol t is not defined'):
+        Model.create('model', statements=Statements() + assign)
+
+    assign = Assignment.create(Expr.function('A', 0), Expr.symbol('0'))
+    with pytest.raises(ValueError, match='Symbol 0 is not defined'):
```

Comment:
This is confusing. Why would we ever name a symbol `0`? If sympy/symengine allows this we should not.

Reply:
I added this test to check that `0` was not added to `symbs_all`. What do you mean with we should not allow it? To create a symbol of an integer or to use a symbol of an integer in statements?

Reply:
I added a PR for this

Reply:
I will remove this test